mne/morph.py

```diff
@@ -0,0 +1,953 @@
+    src_info : dict
+        Information required to transform volume source estimate into
+        Nifti1Image using as_volume
+    morph_data : Trans | dict
+        If morphing a surface source estimate , morph_data is an instance
+        of Trans to morph the data
+        If morphing a volume source estimate morph_data is a dict storing
+        linear and non-linear mapping information.
+    options : **kwargs
+        Keyword arguments for the respective morphing function.
+
+    Notes
+    -----
+    .. versionadded:: X.X.X
+
+    See Also
+    --------
+    X
+
+    """
+    def __init__(self, src, subject_from=None, subject_to='fsaverage',
+                 data_from=None, data_to=None, subjects_dir=None, **options):
+        self.kind = None
+
+        if isinstance(src, string_types):
+            src = read_forward_solution(src)['src']
+
+        # Set attributes
+
+        if isinstance(src, SourceSpaces):
+            self.kind = src.kind
+
+        if subject_from is None:
+            try:
+                subject_from = src[0]['subject_his_id']
+            except KeyError:
+                raise KeyError('subject id in src is None. Please specify'
+                               'subject_from')
+
+        self.subjects_dir = subjects_dir
+        self.subject_from = subject_from
+        self.subject_to = subject_to
+        self.src_info = dict()
+        self.vol_info = dict()
+        self.options = options
+
+        # VolSourceEstimate
+        if self.kind == 'volume':
+
+            if not has_nibabel():
+                raise ImportError(
+                    'NiBabel (Python) must be correctly installed and '
+                    'accessible from Python')
+
+            import nibabel as nib
+
+            logger.info('volume source space inferred...')
+
+            self.vol_info = dict({'mri_from': dict(), 'mri_to': dict()})
+
+            # use subject or manually defined volume
+            mri_subpath = op.join('mri', 'brain.mgz')
+            if data_from is not None:
+                mri_path_from = data_from
+            else:
+                mri_path_from = op.join(self.subjects_dir, self.subject_from,
+                                        mri_subpath)
+            try:
+                logger.info(
+                    'loading ' + mri_path_from + ' as moving volume...')
+                mri_from = nib.load(mri_path_from)
+            except IOError:
+                raise IOError('cannot locate file: ' + mri_path_from)
+
+            if data_to is not None:
+                mri_path_to = data_to
+            else:
+                mri_path_to = op.join(self.subjects_dir, self.subject_to,
+                                      mri_subpath)
+            try:
+                logger.info('loading ' + mri_path_to + ' as static volume...')
+                mri_to = nib.load(mri_path_to)
+            except IOError:
+                raise IOError('cannot locate file: ' + mri_path_to)
```

Comment:
please use formatting string as in the rest of the codebase. Try to copy the patterns from other parts of the code ...